tests/unit/amo/components/TestRatingManager.js

```diff
@@ -190,144 +192,163 @@ describe(__filename, () => {
-      sinon.assert.called(FakeAddonReview);
+    return root
+      .instance()
+      .onSelectRating(5)
+      .then(() => {
+        root.update();
 
-      const props = FakeAddonReview.firstCall.args[0];
-      expect(props.review).toEqual(userReview);
+        expect(root.find(AddonReview)).toHaveLength(1);
 
-      // Now make sure the callback is configured.
-      expect(root.state.showTextEntry).toEqual(true);
-      // Trigger the callback just like AddonReview would
-      // after completion.
-      props.onReviewSubmitted();
-      expect(root.state.showTextEntry).toEqual(false);
-    });
+        // Now make sure the callback is configured.
+        expect(root).toHaveState('showTextEntry', true);
+
+        // Trigger the callback just like AddonReview would after completion.
+        root.instance().onReviewSubmitted();
+
+        expect(root).toHaveState('showTextEntry', false);
+      });
   });
 
   it('calls back to the parent component after submitting a review', () => {
-    const userReview = createInternalReview(fakeReview);
-    const FakeAddonReview = sinon.spy(() => <div />);
     const parentOnReviewSubmitted = sinon.stub();
+    const store = createStoreWithLatestReview();
+
     const root = render({
-      AddonReview: FakeAddonReview,
-      userReview,
       onReviewSubmitted: parentOnReviewSubmitted,
+      store,
     });
 
+    expect(root.find(AddonReview)).toHaveLength(0);
+
     // Select a rating to open the submit review UI.
-    return root.onSelectRating(5).then(() => {
-      sinon.assert.called(FakeAddonReview);
-      const props = FakeAddonReview.firstCall.args[0];
+    return root
+      .instance()
+      .onSelectRating(5)
+      .then(() => {
+        root.update();
 
-      // Simulate pressing submit in the review UI.
-      props.onReviewSubmitted();
+        expect(root.find(AddonReview)).toHaveLength(1);
 
-      // Make sure the parent's callback was executed.
-      sinon.assert.called(parentOnReviewSubmitted);
-    });
+        // Simulate pressing submit in the review UI.
+        root.find(AddonReview).prop('onReviewSubmitted')();
+
+        // Make sure the parent's callback was executed.
+        sinon.assert.called(parentOnReviewSubmitted);
+      });
   });
 
   it('does not render an AddonReview when logged out', () => {
-    const userReview = createInternalReview(fakeReview);
-    const FakeAddonReview = sinon.spy(() => <div />);
-    const userId = null; // logged out
-    const root = render({ AddonReview: FakeAddonReview, userReview, userId });
+    const { store } = dispatchClientMetadata();
 
-    sinon.assert.notCalled(FakeAddonReview);
+    const root = render({ store });
 
-    return root.onSelectRating(5).then(() => {
-      sinon.assert.notCalled(FakeAddonReview);
-    });
+    expect(root.find(AddonReview)).toHaveLength(0);
+
+    return root
+      .instance()
+      .onSelectRating(5)
+      .then(() => {
+        root.update();
```

Comment:
When I remove this the test still passes. Can it be removed? `setState()` changes are applied to the shallow wrapper automatically.